FGEGraphics/UISystem/UIElementText.cs

```diff
@@ -0,0 +1,179 @@
+﻿//
+// This file is part of the Frenetic Game Engine, created by Frenetic LLC.
+// This code is Copyright (C) Frenetic LLC under the terms of a strict license.
+// See README.md or LICENSE.txt in the FreneticGameEngine source root for the contents of the license.
+// If neither of these are available, assume that neither you nor anyone other than the copyright holder
+// hold any right or permission to use this software until such time as the official license is identified.
+//
+
+using System;
+using System.Collections.Generic;
+using FGECore.ConsoleHelpers;
+using FGECore.MathHelpers;
+using FGEGraphics.GraphicsHelpers.FontSets;
+
+namespace FGEGraphics.UISystem;
+
+/// <summary>
+/// A text object that automatically updates its renderable content
+/// based on a <see cref="UIElement"/>'s <see cref="UIElementStyle"/>s.
+/// </summary>
+public class UIElementText
+{
+    /// <summary>The state to display when a required text value is empty.</summary>
+    public const string Null = "null";
+
+    /// <summary>Data internal to a <see cref="UIElementText"/> instance.</summary>
+    public struct InternalData
+    {
+        /// <summary>The parent UI element.</summary>
+        public UIElement ParentElement;
+
+        /// <summary>The raw string content of this text.</summary>
+        public string Content;
+
+        /// <summary>The maximum total width of this text, if any.</summary>
+        public int MaxWidth;
+
+        /// <summary>A cache mapping a UI element's text styles to renderable text.</summary>
+        public Dictionary<UIElementStyle, RenderableText> RenderableContent;
+    }
+
+    /// <summary>The horizontal alignment of the text, if any.</summary>
+    public TextAlignment HorizontalAlignment;
+
+    /// <summary>The vertical alignment of the text, if any.</summary>
+    public TextAlignment VerticalAlignment;
+
+    /// <summary>Whether the text is empty and shouldn't be rendered.</summary>
+    public bool Empty { get; private set; }
```

Comment:
no `private`. Push the value to `InternalData` if you want to prevent it getting mucked with